app/controllers/instructeurs/dossiers_controller.rb

```diff
@@ -517,7 +517,21 @@ def set_gallery_attachments
           .flatten
           .map(&:id)
 
-        champs_attachments_ids + commentaires_attachments_ids
+        avis_attachments_ids = dossier
+          .avis
+          .map { [_1.introduction_file, _1.piece_justificative_file] }
+          .flatten
+          .map(&:attachments)
+          .flatten
+          .compact
+          .map(&:id)
```

Comment:
```suggestion
        avis_attachments_ids = dossier
          .avis.flat_map { [_1.introduction_file, _1.piece_justificative_file] }
          .flat_map(&:attachments)
          .map(&:id)
```

je pense qu'on a pas besoin de compact car `piece_justificative_file.attachments` renvoie `[]` qd il n'y en a pas.

Reply:
👍 je laisse le compact (parfois `flat_map(&:attachments)` renvoie `nil` si on a pas de `introduction_file` ou pas de `piece_justificative_file`). Mais je remplace les `.map.flatten` par `.flat_map`